gui/src/main/java/kcl/teamIndexZero/traffic/gui/SimulationImageProducer.java

```diff
@@ -38,24 +37,30 @@ public SimulationImageProducer(SimulationMap map, BiConsumer<BufferedImage, Simu
 
     }
 
+    private double alpha = 0;
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void tick(SimulationTick tick) {
+        alpha += 0.1;
         graphics.setBackground(Color.WHITE);
         graphics.clearRect(0, 0, image.getWidth(), image.getHeight());
-
-        map.getObjectsOnMap().forEach(object -> {
-            graphics.setColor(object.getColor());
-            MapPosition pos = object.getPosition();
-            graphics.fillRect(
-                    pos.x,
-                    pos.y,
-                    pos.width,
-                    pos.height
-            );
-        });
+        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
+//        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
+        new GUI_Primitives().drawSingleRoad(10, 10, 200, 10, 10, 50, 200, 50, graphics);
```

Comment:
Well in this case I guess I am unable to do both of what you tell me to do. As I said max I can do is stop calling the functions and once incorporated call them in a manner which will not destroy the maps. I am sorry but I have no idea how to set this another way to make it not clear out your current map.                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the EE network.                                                                                                                                                                                                                From: Alexander CherednichenkoSent: Monday 21 March 2016 08:07To: teamIndexZero/index_zero_trafficsystemReply To: teamIndexZero/index_zero_trafficsystemCc: kumar awijeetSubject: Re: [index_zero_trafficsystem] Gui graphics kumar awijeet (#9)In gui/src/main/java/kcl/teamIndexZero/traffic/gui/SimulationImageProducer.java:
>          graphics.setBackground(Color.WHITE);
>          graphics.clearRect(0, 0, image.getWidth(), image.getHeight());
> -
> -        map.getObjectsOnMap().forEach(object -> {
> -            graphics.setColor(object.getColor());
> -            MapPosition pos = object.getPosition();
> -            graphics.fillRect(
> -                    pos.x,
> -                    pos.y,
> -                    pos.width,
> -                    pos.height
> -            );
> -        });
> +        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
> +//        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
> +        new GUI_Primitives().drawSingleRoad(10, 10, 200, 10, 10, 50, 200, 50, graphics);


I don't know of a way in which I can help here, other than removing the calling for my functions, if that helps.
well, that's basically the usual case: either you incorporate your own changes in non-breaking way into the main codebase, or you make them non-breaking, and find someone who incorporates that. 


That's what pull requests are for - basically it is assumed to be a complete piece of work, which can be safely merged to the main branch with button push. No insult here, just explaining how things should normally work. 

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub